UnitTests/UnitTests.UWP/Converters/Test_StringFormatConverter.cs

```diff
@@ -5,59 +5,128 @@
+            var result = converter.Convert(NotEmptyString, typeof(string), null, language);
+
             Assert.AreEqual(NotEmptyString, result);
         }
 
         [TestCategory("Converters")]
         [TestMethod]
-        public void WhenParameterIsTimeFormat_ThenReturnValueOfTimeFormat()
+        [DataRow(null)]
+        [DataRow("en-us")]
+        public void WhenParameterIsInvalidFormat_ThenReturnValue(string language)
         {
             var converter = new StringFormatConverter();
-            var result = converter.Convert(Date, typeof(string), "{0:HH:mm}", "en-us");
-            Assert.AreEqual(Date.ToString("HH:mm"), result);
+            var result = converter.Convert(Date, typeof(string), "{1:}", language);
+
+            Assert.AreEqual(Date, result);
         }
 
         [TestCategory("Converters")]
         [TestMethod]
-        public void WhenParameterIsInvalidFormat_ThenReturnValue()
+        [DataRow(null)]
+        [DataRow("en-us")]
+        public void WhenParameterIsNotAString_ThenReturnValue(string language)
         {
             var converter = new StringFormatConverter();
-            var result = converter.Convert(Date, typeof(string), "{1:}", "en-us");
-            Assert.AreEqual(Date, result);
+            var result = converter.Convert(NotEmptyString, typeof(string), 172, language);
+
+            Assert.AreEqual(NotEmptyString, result);
         }
 
         [TestCategory("Converters")]
         [TestMethod]
-        public void WhenParameterIsNotAString_ThenReturnValue()
+        [DataRow("{0:ddd d MMM}", "ddd d MMM")]
+        [DataRow("{0:HH:mm}", "HH:mm")]
+        [DataRow("{0:hh:mm:ss tt}", "hh:mm:ss tt")]
+        public void WhenValueIsDateTimeAndLanguageIsUnknownAndCultureInfoIsNotSet_ThenReturnValueOfTimeInInvariantCultureFormat(string converterParameter, string expectedFormat)
         {
             var converter = new StringFormatConverter();
-            var result = converter.Convert(NotEmptyString, typeof(string), 172, "en-us");
-            Assert.AreEqual(NotEmptyString, result);
+            var result = converter.Convert(Date, typeof(string), converterParameter, null);
+
+            Assert.AreEqual(Date.ToString(expectedFormat, CultureInfo.InvariantCulture), result);
+        }
+
+        [TestCategory("Converters")]
+        [TestMethod]
+        [DataRow("{0:ddd d MMM}", "ddd d MMM")]
+        [DataRow("{0:HH:mm}", "HH:mm")]
+        [DataRow("{0:hh:mm:ss tt}", "hh:mm:ss tt")]
+        public void WhenValueIsDateTimeAndLanguageIsUnknownAndCultureInfoIsSet_ThenReturnValueOfTimeInConverterCultureInfoFormat(string converterParameter, string expectedFormat)
+        {
+            var converter = new StringFormatConverter();
+            var result = converter.Convert(Date, typeof(string), converterParameter, null);
+
+            Assert.AreEqual(Date.ToString(expectedFormat, CultureInfo.InvariantCulture), result);
+        }
+
+        [TestCategory("Converters")]
```

Comment:
Is there a test we could add like a general date where we have the same format string but then change the language between en/gb and see the result of the date with month/day flipped? Then we know for sure it worked, right?

A lot of the current test cases just seem to be testing `string.Format` to `ToString` in a way we know should work, right? (which is important, but not really replicating how we expect it to be used in an app, right?)

Reply:
@Arlodotexe you could have left them, was just suggesting an additional test, looks like something was missed in the clean-up:

```
         D:\a\1\s\UnitTests\UnitTests.UWP\Converters\Test_StringFormatConverter.cs(19,41): error CS0414: The field 'Test_StringFormatConverter.Amount' is assigned but its value is never used [D:\a\1\s\UnitTests\UnitTests.UWP\UnitTests.UWP.csproj]
```